poc/vdaf_poc/field.py

```diff
@@ -272,6 +264,103 @@ def poly_eval(field: type[F], p: list[F], eval_at: F) -> F:
 
 def poly_interp(field: type[F], xs: list[F], ys: list[F]) -> list[F]:
     """Compute the Lagrange interpolation polynomial for the given points."""
-    R = PolynomialRing(field.gf, 'x')
-    p = R.lagrange_polynomial([(x.val, y.val) for (x, y) in zip(xs, ys)])
-    return poly_strip(field, list(map(field, p.coefficients())))
+
+    # This uses Newton's divided difference interpolation formula.
+
+    assert len(xs) == len(ys)
+    n = len(xs)
+    one = field(1)
+
+    # We interleave three computations in each iteration of the outermost loop.
+    # First, we compute the ith Newton basis polynomial. Second, we compute the
```

Comment:
nit: I find adding backticks around math improves readability. Just a suggestion, feel free to ignore. (Here and below.)
```suggestion
    # First, we compute the `i`th Newton basis polynomial. Second, we compute the
```